x-pack/plugins/maps/public/components/validated_number_input.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Component, ChangeEvent, MouseEvent } from 'react';
+import { EuiFieldNumber, EuiFormRow } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import _ from 'lodash';
+
+interface State {
+  value: number | string;
+  errorMessage: string;
+  isValid: boolean;
+}
+
+interface Props {
+  initialValue: number;
+  min: number;
+  max: number;
+  onChange: (value: number) => void;
+  label: string;
+}
+
+function isNumberValid(value: number | string, min: number, max: number) {
+  const parsedValue = parseInt(value.toString(), 10);
```

Comment:
What about floats? I think this input could be used in a number of cases so instead of using parseInt, use parseFloat.